lib/foreman/renderer/scope/macros/helpers.rb

```diff
@@ -11,6 +11,33 @@ module Helpers
             sections only: %w[all]
           end
 
+          apipie :method, "Returns counter increased by given value (as String) stored in a hostgroup param named 'sequence_num_xxx'" do
+            required :hostgroup_name, String, "hostgroup title to store in (e.g. RHEL7/base)"
+            optional :increment, Integer, "integer increment, can be negative too", default: 1
+            optional :name, String, "name of the sequence (prefix for hostgroup parameter)", default: 'default'
+            optional :prefix, Integer, "leading zeroes", default: 1
+            returns String, "String with incremented value or empty string if hostgroup not found"
+            example 'sequence_hostgroup_param_next("myHG", 10) #=> 10'
+            example 'sequence_hostgroup_param_next("myHG", 2, "kiosk", 5) #=> 00002'
+          end
+          def sequence_hostgroup_param_next(hostgroup_name, increment = 1, name = "default", prefix = 1)
+            hostgroup = Hostgroup.find_by_title(hostgroup_name)
```

Comment:
Could a user use this to access hostgroups they shouldn't?

Reply:
Good point, they can only access the parameter, which is I believe purpose of this helper. Ideally this should be a host helper, but the user actually needs this for discovery naming where host is not yet available (hostgroup was not yet assigned). I'd say we actually waive this because the only thing an attacker could do would be increasing a counter - he'd need to know the hostgroup name and counter name in advance. I am trying to figure out if this is a problem.